webpack/assets/javascripts/react_app/routes/RegistrationCommands/RegistrationCommandsPage/components/fields/OperatingSystem.js

```diff
@@ -51,7 +52,9 @@ const OperatingSystem = ({
       handleOperatingSystem('');
       dispatch(operatingSystemTemplateAction(hostGroupOsId));
     }
-  }, [dispatch, handleOperatingSystem, hostGroupId, hostGroups]);
+    // Do not use hostGroups & handleOperatingSystem as useEffect dependency
+    // it will cause a bug when Host group has OS
+  }, [dispatch, hostGroupId]);
```

Comment:
This sounds like something odd is happening as part of this hook, perhaps it's worth investigating further to check if there's some hidden edge case here or something different than should be done?

Reply:
Looks like the error is not related to my changes, its also on rh_cloud: https://github.com/theforeman/foreman_rh_cloud/pull/536/checks?check_run_id=2437872203#step:7:7